Fix channel alarm state detection at startup for FXO channels (#______)

Upstream issue: http://bugs.digium.com/view.php?id=12160
Source: http://bugs.digium.com/file_download.php?file_id=19277&type=bug

--- a/channels/chan_zap.c
+++ b/channels/chan_zap.c
@@ -7429,6 +7429,9 @@ static struct zt_pvt *mkintf(int channel
 				}
 				tmp->law = p.curlaw;
 				tmp->span = p.spanno;
+#ifdef ZT_GET_PARAMS_V1
+				tmp->inalarm = (p.chan_alarms != 0);
+#endif
 				span = p.spanno - 1;
 			} else {
 				if (channel == CHAN_PSEUDO)
@@ -7796,10 +7799,12 @@ static struct zt_pvt *mkintf(int channel
 			ioctl(tmp->subs[SUB_REAL].zfd,ZT_SETTONEZONE,&tmp->tonezone);
 #ifdef HAVE_PRI
 			/* the dchannel is down so put the channel in alarm */
-			if (tmp->pri && !pri_is_up(tmp->pri))
-				tmp->inalarm = 1;
-			else
-				tmp->inalarm = 0;
+			if (tmp->pri) {
+				if (!pri_is_up(tmp->pri))
+					tmp->inalarm = 1;
+				else
+					tmp->inalarm = 0;
+			}
 #endif				
 			memset(&si, 0, sizeof(si));
 			if (ioctl(tmp->subs[SUB_REAL].zfd,ZT_SPANSTAT,&si) == -1) {
